### PR TITLE
[ace] Update to 7.0.6

### LINKED ISSUE #21525: [ace] build failure
**Host Environment**
- OS: WSL (Ubuntu 20.04.3 LTS (GNU/Linux 4.4.0-19041-Microsoft x86_64)
- Compiler: g++ (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0

**To Reproduce**
`./vcpkg install ace[tao,ssl,wchar,zlib]`

**Failure logs** 
```
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:155 (message):
    Command failed: make ssl=1 debug=1 optimize=0 -j9
    Working Directory: /mnt/d/vcpkg/buildtrees/ace/src/ACE-TAO-7-f61ead9ce9.clean/TAO
    See logs for more information:
      /mnt/d/vcpkg/buildtrees/ace/make-x64-linux-dbg-out.log
      /mnt/d/vcpkg/buildtrees/ace/make-x64-linux-dbg-err.log

Call Stack (most recent call first):
  ports/ace/portfile.cmake:347 (vcpkg_execute_build_process)
  scripts/ports.cmake:142 (include)
```
[mwc-x64-linux-out.log](https://github.com/microsoft/vcpkg/files/7566386/mwc-x64-linux-out.log)
[make-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/7566383/make-x64-linux-dbg-out.log)
[make-x64-linux-dbg-err.log](https://github.com/microsoft/vcpkg/files/7566384/make-x64-linux-dbg-err.log)

**Additional context**
vcpkg HEAD: `cbc22a396c88d098247fe7f79f0913a952e1f77d`
When I downgrade to `8831ef24eee7e04009fe464097680512e46138a4` , I get an error inside `vcpkg_fixup_pkgconfig`.

### PR DESCRIPTION
**Pull request by Cheney-W**
**Describe the pull request**
 
  Fixes https://github.com/microsoft/vcpkg/issues/21525
  Remove the expired patch.

### CODE REVIEW

**Comment by jwillemsen** [comment]
Patch for macosx could maybe be removed completely

File: ports/ace/portfile.cmake [hunk 1]
```diff
@@ -25,22 +25,20 @@ else()
...
         FILENAME ACE-src-${ACE_VERSION}.tar.gz
-        SHA512 6e5e43e600763e612c292cb88443e4dce1be94d049e2a784c5a4d4720314b484cccec4f7f05534c6ef824d86bef8bfe4ff5bce5f7998896cdaa599302b5b2562
+        SHA512 9770fab3552835803a93c9a234218c9dd961ecde67227ee92e0972cd2e2ff267147b255ab437453a887bc47b20f70c7a64efeada5dde5d3ea2cade54200e8354
     )
   endif()
 endif()
 
 vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
     ARCHIVE ${ARCHIVE}
```
Comment (JackBoosY) [inline ports/ace/portfile.cmake:41]:
```suggestion
    ARCHIVE “${ARCHIVE}”
```
Please also add double quotes to the paths below.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by jwillemsen** [comment]
> You have modified or added at least one vcpkg.json where a "license" field is missing. If you feel able to do so, please consider adding a "license" field to the following files:
> 
>     * `ports/ace/vcpkg.json`
> 
> 
> Valid values for the license field are listed at https://spdx.org/licenses/

As license DOC should be used (see https://spdx.org/licenses/DOC.html)

File: ports/ace/portfile.cmake [hunk 2]
```diff
@@ -103,15 +101,15 @@ if(VCPKG_TARGET_IS_WINDOWS)
   if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
 	set(MPC_STATIC_FLAG -static)
   endif()
-  file(WRITE ${ACE_SOURCE_PATH}/config.h "#include \"ace/config-windows.h\"")
+  file(WRITE "${ACE_SOURCE_PATH}/config.h" "#include \"ace/config-windows.h\"")
```
Comment (Osyotr) [inline ports/ace/portfile.cmake:104]: Would you kindly also include this change? Just replace lines 104-113 with below code:
```cmake
  set(config_h_contents "#include \"ace/config-windows.h\"\n")
elseif(VCPKG_TARGET_IS_LINUX)
  set(SOLUTION_TYPE gnuace)
  set(config_h_contents "#include \"ace/config-linux.h\"\n")
  file(WRITE "${ACE_ROOT}/include/makeinclude/platform_macros.GNU" "include $(ACE_ROOT)/include/makeinclude/platform_linux.GNU")
elseif(VCPKG_TARGET_IS_OSX)
  set(SOLUTION_TYPE gnuace)
  set(config_h_contents "#include \"ace/config-macosx.h\"\n")
  file(WRITE "${ACE_ROOT}/include/makeinclude/platform_macros.GNU" "include $(ACE_ROOT)/include/makeinclude/platform_macosx.GNU")
endif()

if("wchar" IN_LIST FEATURES)
  string(APPEND config_h_contents "#define ACE_USES_WCHAR 1\n")
endif()
file(WRITE "${ACE_SOURCE_PATH}/config.h" "${config_h_contents}")
```
```

Comment (Cheney-W) [reply ports/ace/portfile.cmake]: Done!

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the suggestion @jwillemsen!

**Pull request merged**